src/presence_analyzer/tests.py

```diff
@@ -85,6 +147,41 @@ def test_get_data(self):
             data[10][sample_date]['start'],
             datetime.time(9, 39, 5)
         )
+        self.assertTrue(mock_log.debug.called)
+
+    def test_mean(self):
+        """
+        Test mean() function.
+        """
+        mean = utils.mean
+        self.assertEqual(0, mean([]))
+        self.assertEqual(3, mean([1, 2, 3, 4, 5]))
+
+    def test_interval(self):
+        """
+        Test interval() function.
+        """
+        interval = utils.interval
+        start = datetime.time(11, 30, 0)
+        end = datetime.time(12, 30, 0)
+        self.assertEqual(3600, interval(start, end))
+
+    def test_seconds_since_midnight(self):
+        """
+        Test seconds_since_midnight() function.
+        """
+        calc_sec = utils.seconds_since_midnight
+        time = datetime.time(1, 1, 1)
+        self.assertEqual(3661, calc_sec(time))
+
+    def test_group_by_weekday(self):
+        """
+        Test assigning time to weekdays
+        """
+        group_by_weekday = utils.group_by_weekday
+        data = utils.get_data()
```

Comment:
Let's not make this dependent on a CSV of any sort.
Create an in-place test object and then pass it to grouping func.